### ISSUE TITLE
Vector  concat

### ISSUE THREAD

**Issue by sleexyz** [sleexyz/hylogen#61]
```
(:+) :: forall (n :: Nat) (m :: Nat). (Veccable n, Veccable m) 
     => Vec n -> Vec m -> Vec (n + m)
```

**Comment by mdaiter** [comment]
Like this? http://docs.idris-lang.org/en/latest/tutorial/typesfuns.html#vectors

+1 for this feature. Hit me up if you want to work on this together (I'm getting more into Idris/Agda programming and category theory too these days :) I'd love to help out. )

**Comment by sleexyz** [comment]
yeah! it should be fairly easy. There just needs to be a way to reify a term from a 'Nat' at the type level, maybe with typeclasses? Perhaps i already have the mechanisms in place, haven't touch this project in a month. But feel free to poke around, let me know if you have questions!

(yeah, getting into Agda!)
